lib/app/modules/settings/views/weather_api.dart

```diff
@@ -33,124 +33,175 @@ class WeatherApi extends StatelessWidget {
           title: 'API Key',
           titleStyle: Theme.of(context).textTheme.displaySmall,
           content: Obx(
-            () => Container(
-              // If the user hasn't clicked on the 'Save' or 'Update' button of the dialog, or there is no error in the weather key, then show the dialog
-              child: (controller.weatherKeyState.value !=
-                          WeatherKeyState.saveAdded &&
-                      controller.weatherKeyState.value !=
-                          WeatherKeyState.saveUpdated)
-                  ? (controller.didWeatherKeyError.value == false)
-                      ? Column(
-                          children: [
-                            TextField(
-                              obscureText: false,
-                              controller: controller.apiKey,
-                            ),
-                            Padding(
-                              padding: const EdgeInsets.only(top: 20.0),
-                              child: Row(
-                                mainAxisAlignment:
-                                    MainAxisAlignment.spaceEvenly,
-                                children: [
-                                  TextButton(
-                                      style: ButtonStyle(
-                                        backgroundColor:
-                                            MaterialStateProperty.all(
-                                                kprimaryColor),
-                                      ),
-                                      child: Text(
-                                        // If the weather state is add, then show 'Save' else show 'Update' text on the button
-                                        controller.weatherKeyState.value ==
-                                                WeatherKeyState.add
-                                            ? 'Save'
-                                            : 'Update',
-                                        style: Theme.of(context)
-                                            .textTheme
-                                            .displaySmall!
-                                            .copyWith(
-                                                color: themeController
-                                                        .isLightMode.value
-                                                    ? kLightPrimaryTextColor
-                                                    : ksecondaryTextColor),
-                                      ),
-                                      onPressed: () async {
-                                        Utils.hapticFeedback();
+            () => Stack(
```

Comment:
Any specific reason you went with Stack?

Reply:
I used Stack because I wanted to have that `CircularProgressIndicator` on the top layer above the dialog.